src/webui/webapplication.cpp

```diff
@@ -400,13 +400,25 @@ void WebApplication::action_command_download()
     params.savePath = savepath;
     params.category = category;
 
+    bool firstTorrent = true;
+    bool allTorrentsAdded = false;
+
     foreach (QString url, list) {
         url = url.trimmed();
         if (!url.isEmpty()) {
             Net::DownloadManager::instance()->setCookiesFromUrl(cookies, QUrl::fromEncoded(url.toUtf8()));
-            BitTorrent::Session::instance()->addTorrent(url, params);
+            bool addSucceeded = BitTorrent::Session::instance()->addTorrent(url, params);
+
+            if (addSucceeded && firstTorrent)
+                allTorrentsAdded = true;
+            firstTorrent = false;
```

Comment:
This could easily turn into a partial success vs partial failure debate, so I'll defer to your judgement for the sake of getting this added ;)